book/src/traits/traits.md

```diff
@@ -49,15 +65,25 @@ Instead entries must be deleted and newly stored.
 {{#include ../../../traits/src/key_store.rs:15:40}}
 ```
 
+**NOTE:** Right now key material need to be extractable from the key store.
+This will most likely change in future.
+
 ### OpenMlsCryptoProvider
 
 Additionally, there's a wrapper trait defined that is expected to be passed into
 the public OpenMLS API.
+Some OpenMLS APIs require only one of the sub-traits though.
 
 ```rust,no_run,noplayground
 {{#include ../../../traits/src/traits.rs:15:28}}
 ```
 
+## Implementation Notes
+
+It is not necessary to implement all sub-traits if one functionality is missing.
+If you want to use a persisting key store for example, it is sufficient to do a new implementation of the key store trait and combine it with one of the provided crypto and randomness trait implementations.
```

Comment:
Would that be a reason to make the `memory_keystore` public again?

Reply:
🤔  it is a published crate. So how do you think it is not public?

Reply:
Never mind. I got two things confused, I think.